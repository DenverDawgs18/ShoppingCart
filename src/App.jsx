
import "./App.css";
import Navbar from "./Navbar";

function App() {
  return (
    <>
      <Navbar />
      <div className="homepage">
        <h1>Homepage</h1>
        
      </div>
      
    </>
  );
}

export default App;
