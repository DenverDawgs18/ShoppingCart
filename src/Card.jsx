import { useEffect, useState } from "react";

export default function Card({id}){
    const [obj, setObj] = useState(null)
    console.log(id)
useEffect(() => {
    fetch(`https://fakestoreapi.com/products/` + id, {
        mode: 'cors'
    })
    .then((res) => res.json())
    .then((json) => setObj(json))
})
    console.log(obj)
    return(
        obj && ( <div className="card">
            <img src={obj.image} alt="ima" className="cardImage" />
            <div className="bottom">
                <h3>{obj.title}</h3>
                <input type="number" name="quantity" id="quantity" defaultValue={1} />
                <button type="submit" className="add">Add to cart</button>
            </div>
            
        </div>
    ))
}