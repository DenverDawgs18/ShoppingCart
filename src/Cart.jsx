import Navbar from "./Navbar"
import {ShopContext}  from "./Router"
import {useContext} from 'react';
import { Link } from "react-router-dom";
import Icon from "@mdi/react";
import {mdiMinusCircle, mdiPlusBox} from "@mdi/js"
export default function Cart(){
    const {cartItems, removeFromCart, changeQuantity, total} = useContext(ShopContext)
    
   if(cartItems.length !== 0){

   
    return(
        <>
            <Navbar />
           <div className="cart">
                <div className="cartCards">

                    <h1 className="your">Your cart</h1>
                    {cartItems.map(item => (
                        <div className="cartCard" key={item.id}>
                            <img src={item.image} alt="ima" className="cardImage cartImage" />       
                            <div className="cartCardRight">
                                <div className="oneLine">
                                    <h3>{item.title}</h3>
                                    <button onClick={() => removeFromCart(item)} className="remove">X</button>
                                </div>
                                <h4>{item.quantity > 1 ? '$' + item.quantity * item.price + 
                                "  ($" + item.price + ' per)' : '$' + item.price}</h4>
                
                                <div className="quant">
                                    <Icon path={mdiMinusCircle} size={1} onClick={() => changeQuantity(item, 'minus')} />
                                    <h4>Quantity: {item.quantity}</h4>
                                    <Icon path={mdiPlusBox} size={1} onClick={() => changeQuantity(item, 'plus')}/>
                                </div>
                            </div>
                        </div>
                    ))}
                </div>
                <div className="cartRight">
                    <h1>{total ? 'Total: $' + total 
                    : ''}</h1>
                    <button className="proceed">PROCEED TO CHECKOUT</button>
                    <Link to="/products" style={{textDecoration: 'none', color: 'black'}}><h3 className="addp">ADD PRODUCTS</h3></Link>
                </div>
            </div>
        </>
    )
    }
    else{
        return(
            <>
            <Navbar />
            <div className="none">
                
                <Link to="/products" style={{textDecoration: 'none', color: 'black'}}><h3 className="addp">ADD PRODUCTS</h3></Link>
            </div>
            

            </>
        )
    }
}