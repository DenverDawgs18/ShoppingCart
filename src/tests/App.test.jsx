import { describe, it, expect } from "vitest";
import App from '../App.jsx';
import Cart from '../Cart.jsx';
import Card from '../Card.jsx'
import {render} from '@testing-library/react'
import { BrowserRouter } from "react-router-dom";

describe('App component', () => {
    it('renders against snapshot', () => {
        const {wrapper} = render(
        <BrowserRouter>
            <App />
        </BrowserRouter>
        );
        expect(wrapper).toMatchSnapshot()
    })
})
describe('Cart component', () => {
    it('renders against snapshot', () => {
        const {wrapper} = render(
        <BrowserRouter>
            <Cart />
        </BrowserRouter>
        );
        expect(wrapper).toMatchSnapshot()
    })
})
describe('Card component', () => {
    it('renders against snapshot', () => {
        const {wrapper} = render(
        <BrowserRouter>
            <Card />
        </BrowserRouter>
        );
        expect(wrapper).toMatchSnapshot()
    })
})