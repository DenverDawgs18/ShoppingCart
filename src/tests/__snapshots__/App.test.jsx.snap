// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[` component > renders against snapshot 1`] = `undefined`;

exports[`App component > renders 1`] = `undefined`;

exports[`App component > renders against snapshot 1`] = `undefined`;

exports[`Card component > renders against snapshot 1`] = `undefined`;

exports[`Cart component > renders against snapshot 1`] = `undefined`;

exports[`Cart component > renders against snapshot 2`] = `undefined`;
